javaslang/src/test/java/javaslang/collection/AbstractTraversableTest.java

```diff
@@ -1264,9 +1264,15 @@ public void shouldCaclNonemptyOrElseOther() {
     }
 
     @Test
-    public void shouldCaclEmptyOrElseSupplier() {
-        Supplier<Iterable<Integer>> other = () -> of(42);
-        assertThat(empty().orElse(other)).isEqualTo(of(42));
+    public void shouldCaclEmptyOrElseSameSupplier() {
+        Iterable<Integer> other = of(42);
+        Supplier<Iterable<Integer>> supplier = () -> other;
+        assertThat(empty().orElse(supplier)).isSameAs(other);
```

Comment:
also adds test to check that `orElse()` result is same as result of evaluating supplier in case of properly iterable type